### PR TITLE
Refactor MenuItemMorph, PluggableMenuItemSpec and PragmaMenuAndShortcutRegistrationItem to support use of a FormSet and apply that to CmdCommand

### PR DESCRIPTION
**Pull request by Rinzwind**
This pull request refactors MenuItemMorph to support use of a FormSet for the icon and applies that to the CmdCommand hierarchy. In the CmdCommand hierarchy, the overrides of `#defaultMenuIcon` are refactored to override `#defaultMenuIconName` instead.

### CODE REVIEW

**Comment by Rinzwind** [comment]
I converted this to a draft. Commit 02f48dcb43a0cc8d introduces a bug in `#setUpIconForMenuItem:` where the argument, despite the variable name ‘aMenuItemMorph’, can also be a PragmaMenuAndShortcutRegistration, which does not understand `#iconFormSet:`.

**Comment by Rinzwind** [comment]
I fixed the bug referred to in my previous message by first refactoring PluggableMenuItemSpec and PragmaMenuAndShortcutRegistrationItem as well.

File: src/MenuRegistration/PragmaMenuAndShortcutRegistrationItem.class.st [hunk 1]
```diff
@@ -13,6 +13,7 @@ Class {
 		'label',
 		'help',
 		'icon',
```
Comment (jecisc) [inline src/MenuRegistration/PragmaMenuAndShortcutRegistrationItem.class.st]:
```suggestion
```

Comment (jecisc) [reply src/MenuRegistration/PragmaMenuAndShortcutRegistrationItem.class.st]: This instance variable is not used anymore and should be removed

**Review by jecisc — CHANGES_REQUESTED** [review]

File: src/MenuRegistration/PragmaMenuAndShortcutRegistrationItem.class.st (hunk 1 above)
Comment (Rinzwind) [reply src/MenuRegistration/PragmaMenuAndShortcutRegistrationItem.class.st]: Forgot to actually remove it indeed, fixed in commit 49a7a887efa18e50.

**Comment by Ducasse** [comment]
@jecisc can you check it? Tx

**Comment by Rinzwind** [comment]
@jecisc Could I ask for an update on the (re)review? If you’re not available, it might be better to let someone else take over?

**Review by jecisc — APPROVED** [review]

**Pull request merged**